### ISSUE TITLE
Simple authentication with v8

### ISSUE THREAD

**Issue by IsaackRasmussen** [cosullivan/SmtpServer#149]
I'm using version v8.04 and I can't get it to advertise AUTH and my mail client claims its not supported

```
                        X509Certificate2 cert = new X509Certificate2("./properties/certificate.pfx","password");

                        services.AddTransient<IMessageStore, SmtpMessageStore>();
                        services.AddTransient<IUserAuthenticator, SmtpUserAuthenticator>();

                        services.AddSingleton(
                            provider =>
                            {
                                var options = new SmtpServerOptionsBuilder()
                                    .ServerName("SMTP")
                                    .Endpoint(endpointBuilder => endpointBuilder
                                            .Port(587)
                                            .IsSecure(false)
                                            .AllowUnsecureAuthentication(true)
                                            .AuthenticationRequired(true)
                                            .Build())
                                    .Certificate(cert)
                                    .Build();


                                return new SmtpServer.SmtpServer(options, provider.GetRequiredService<IServiceProvider>());
                            });

                        services.AddHostedService<SmtpReceiverService>();

```

Is a certificate required, even when its not secure? What else is required to make it advertise AUTH?

**Comment by cosullivan** [comment]
Hi @IsaackRasmussen , 

I think the issue might be here;
https://github.com/cosullivan/SmtpServer/blob/master/Src/SmtpServer/Protocol/EhloCommand.cs#L84

It checks to see if you have an IUserAuthenticatorFactory registered, but in theory it should be testing for either IUserAuthenticatorFactory or IUserAuthenticator. 

On your SmtpUserAuthenticator class can you also implement IUserAuthenticatorFactory and just make the CreateInstance method "return this" and see if that works?

You would also need to register it to.
`services.AddTransient<IUserAuthenticatorFactory, SmtpUserAuthenticator>();`

I will fix this in the next few days assuming this is the issue.

Thanks, Cain.

**Comment by IsaackRasmussen** [comment]
Hi Cain,

Thanks... that worked. I had already tried implementing the Factory but did not register it - which did the trick.

**Comment by cosullivan** [comment]
Hi, this is now resolved in v8.0.5.
